File: frontend/src/types/cpValidation.ts
export interface ValidationResult {
  occurrence_id: string;
  run_id: string;
  finding_id: string;
  cp_id: string;
  validation_type: string;
  rule_id: string;
  severity: "error" | "warning" | "info";
  category: string;
  title: string;
  description: string | null;
  affected_items: string[];
  fmea_node_ids: string[];
  suggestion: string | null;
  suggestion_data: Record<string, unknown> | null;
  status: "open" | "accepted" | "rejected" | "resolved";
  resolved_by: string | null;
  resolved_at: string | null;
  present: boolean;
  created_at: string;
}

export interface ValidationRun {
  run_id: string;
  cp_id: string;
  trigger: string;
  status: "running" | "completed" | "failed";
  rule_count: number;
  error_count: number;
  warning_count: number;
  info_count: number;
  started_at: string;
  completed_at: string | null;
  failed_rules: unknown[];
  created_by: string | null;
}

export interface ValidationSummary {
  run_id: string | null;
  status: string | null;
  total: number;
  error_count: number;
  warning_count: number;
  info_count: number;
  open_count: number;
  resolved_count: number;
  rejected_count: number;
}

export interface ValidationResultsList {
  items: ValidationResult[];
  total: number;
}

File: backend/app/schemas/cp_validation.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class ValidationFindingResponse(BaseModel):
    finding_id: uuid.UUID
    cp_id: uuid.UUID
    finding_hash: str
    rule_id: str
    severity: str
    category: str
    status: str
    resolved_by: uuid.UUID | None = None
    resolved_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ValidationOccurrenceResponse(BaseModel):
    occurrence_id: uuid.UUID
    run_id: uuid.UUID
    finding_id: uuid.UUID
    cp_id: uuid.UUID
    validation_type: str
    title: str
    description: str | None = None
    affected_items: list = []
    fmea_node_ids: list = []
    suggestion: str | None = None
    suggestion_data: dict | None = None
    present: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ValidationResultItem(BaseModel):
    """Joined occurrence + finding, returned by list endpoint."""
    occurrence_id: uuid.UUID
    run_id: uuid.UUID
    finding_id: uuid.UUID
    cp_id: uuid.UUID
    validation_type: str
    rule_id: str
    severity: str
    category: str
    title: str
    description: str | None = None
    affected_items: list = []
    fmea_node_ids: list = []
    suggestion: str | None = None
    suggestion_data: dict | None = None
    status: str
    resolved_by: uuid.UUID | None = None
    resolved_at: datetime | None = None
    present: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ValidationRunResponse(BaseModel):
    run_id: uuid.UUID
    cp_id: uuid.UUID
    trigger: str
    status: str
    rule_count: int
    error_count: int
    warning_count: int
    info_count: int
    started_at: datetime
    completed_at: datetime | None = None
    failed_rules: list = []
    created_by: uuid.UUID | None = None

    model_config = {"from_attributes": True}


class ValidationSummaryResponse(BaseModel):
    run_id: uuid.UUID | None = None
    status: str | None = None
    total: int = 0
    error_count: int = 0
    warning_count: int = 0
    info_count: int = 0
    open_count: int = 0
    resolved_count: int = 0
    rejected_count: int = 0


class ValidationResultsListResponse(BaseModel):
    items: list[ValidationResultItem]
    total: int
